lib/cosmos/dart/script/dart_decom_server.rb

```diff
@@ -0,0 +1,201 @@
+    Cosmos::Logger.info("#{request_start_time.formatted}: QUERY: #{request}")
+
+    begin
+      start_time_sec = request['start_time_sec']
+      start_time_usec = request['start_time_usec']
+      end_time_sec = request['end_time_sec']
+      end_time_usec = request['end_time_usec']
+
+      start_time = nil
+      end_time = nil
+      start_time = Time.at(start_time_sec, start_time_usec) if start_time_sec and start_time_usec
+      end_time = Time.at(end_time_sec, end_time_usec) if end_time_sec and end_time_usec
+
+      item = request['item']
+      raise "Item \"#{item}\" invalid" if item.length != 3
+
+      reduction = request['reduction'].to_s.upcase
+      case reduction
+      when "", "NONE"
+        reduction = :NONE
+        reduction_modifier = ""
+      when "MINUTE"
+        reduction = :MINUTE
+        reduction_modifier = "_m"
+      when "HOUR"
+        reduction = :HOUR
+        reduction_modifier = "_h"
+      when "DAY"
+        reduction = :DAY
+        reduction_modifier = "_d"
+      else
+        raise "Unknown reduction: #{reduction}"
+      end
+
+      value_type = request['value_type'].to_s.upcase
+      case value_type.upcase
+      when 'RAW'
+        not_value_type = ItemToDecomTableMapping::CONVERTED
+        item_name_modifier = ""
+        raise "RAW value_type is only valid with NONE reduction" if reduction != :NONE
+      when 'RAW_MAX'
+        not_value_type = ItemToDecomTableMapping::CONVERTED
+        item_name_modifier = "max"
+        raise "RAW_MAX value_type is not valid with NONE reduction" if reduction == :NONE
+      when 'RAW_MIN'
+        not_value_type = ItemToDecomTableMapping::CONVERTED
+        item_name_modifier = "min"
+        raise "RAW_MIN value_type is not valid with NONE reduction" if reduction == :NONE
+      when 'RAW_AVG'
+        not_value_type = ItemToDecomTableMapping::CONVERTED
+        item_name_modifier = "avg"
+        raise "RAW_AVG value_type is not valid with NONE reduction" if reduction == :NONE
+      when 'CONVERTED'
+        not_value_type = ItemToDecomTableMapping::RAW
+        item_name_modifier = ""
+        raise "CONVERTED value_type is only valid with NONE reduction" if reduction != :NONE
+      when 'CONVERTED_MAX'
+        not_value_type = ItemToDecomTableMapping::RAW
+        item_name_modifier = "max"
+        raise "CONVERTED_MAX value_type is not valid with NONE reduction" if reduction == :NONE
+      when 'CONVERTED_MIN'
+        not_value_type = ItemToDecomTableMapping::RAW
+        item_name_modifier = "min"
+        raise "CONVERTED_MIN value_type is not valid with NONE reduction" if reduction == :NONE
+      when 'CONVERTED_AVG'
+        not_value_type = ItemToDecomTableMapping::RAW
+        item_name_modifier = "avg"
+        raise "CONVERTED_AVG value_type is not valid with NONE reduction" if reduction == :NONE
+      else
+        raise "Unknown value_type: #{value_type}"
+      end
+
+      meta_ids = request['meta_ids']
```

Comment:
meta_ids appear to be the PacketLogEntry database IDs. Is that correct? How does the user of this API know the meta_ids?

Reply:
That's correct.   You have to first query something about the SYSTEM META packet that you care about.  For example you could get SYSTEM META SERIAL_NUMBER and then note the meta packets that have the serial number you care about.

Reply:
Can you create a concrete example of this in your dart_decom_client script because I'm still confused.